### PR TITLE
Validate groups with roles before creating request

### PR DESCRIPTION
**Pull request by hsong-rh**
This PR added a logic to make sure that the groups used to create a request should have approval roles, otherwise will raise an exception.

### CODE REVIEW

File: app/models/group.rb [hunk 1]
```diff
@@ -22,6 +22,15 @@ def self.all(username = nil)
     groups
   end
 
+  def with_approval_role?
```
Comment (mkanoor) [inline app/models/group.rb]: @hsong-rh Could this be redone based on this issue https://github.com/RedHatInsights/insights-rbac/issues/115
To use the role names directly

Comment (hsong-rh) [reply app/models/group.rb]: I don't think so, RedHatInsights/insights-rbac#115 will return roles based on users. Here I need check specific approval roles for groups.

File: app/services/jbpm_process_service.rb [hunk 2]
```diff
@@ -39,6 +39,12 @@ def process_options
     options
   end
 
+  def validate(group)
+    raise Exceptions::RBACError, "Group #{group.uuid} doesn't have needed roles" unless group.with_approval_role?
```
Comment (bzwei) [inline app/services/jbpm_process_service.rb]: Can we print group name (or both) instead? The text can be straight `Group does not have approval role".

**Comment by bzwei** [comment]
@hsong-rh can this PR be refactored with the new rbac api features?

**Comment by bzwei** [comment]
not valid because of #243

**Pull request closed without merging**